mipengine/controller/node_landscape_aggregator.py

```diff
@@ -208,33 +212,31 @@ def stop(self):
             self._keep_updating = False
             self._update_loop_thread.join()
 
-    def set_new_registy_values(self, nodes, data_models, datasets_locations):
-        _log_node_changes(self._registries.node_registry.nodes, nodes)
+    def set_new_registries(self, node_registry, data_model_registry):
+        _log_node_changes(
+            self._registries.node_registry.get_nodes(),
+            node_registry.get_nodes(),
+        )
         _log_data_model_changes(
             self._registries.data_model_registry.data_models,
-            data_models,
+            data_model_registry.data_models,
         )
         _log_dataset_changes(
             self._registries.data_model_registry.datasets_locations,
-            datasets_locations,
-        )
-        _node_registry = NodeRegistry(nodes=nodes)
-        _data_model_registry = DataModelRegistry(
-            data_models=data_models,
-            datasets_locations=datasets_locations,
+            data_model_registry.datasets_locations,
         )
         self._registries = _NLARegistries(
-            node_registry=_node_registry, data_model_registry=_data_model_registry
+            node_registry=node_registry, data_model_registry=data_model_registry
         )
 
-    def get_nodes(self) -> Dict[str, NodeInfo]:
-        return self._registries.node_registry.nodes
+    def get_nodes(self) -> List[NodeInfo]:
+        return self._registries.node_registry.get_nodes()
 
     def get_global_node(self) -> NodeInfo:
-        return self._registries.node_registry.get_global_node()
+        return self._registries.node_registry.global_node
```

Comment:
The `NodeRegistry` could have more than one potential `global_node` and here you could get the first one only or fail if none exists.